src/components/ParksWrapper/ParksWrapper.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import './ParksWrapper.css';
+
+interface ParksWrapperProps {
+  park: {
+    id: string;
+    fullName: string;
+    description: string;
+    activities: Array<{ id: string; name: string }>;
+  } | null;
+}
+
+function ParksWrapper(props: ParksWrapperProps) {
+  const { park } = props;
+
+  if (!park) {
+    return <div className="error">Park information is not available.</div>;
+  }
+
+  if (!park.activities || park.activities.length === 0) {
+    return (
+      <div className="parks-wrapper">
+        <h1>{park.fullName}</h1>
+        <p>{park.description}</p>
+        <h2>Activities</h2>
+        <p>No activities available for this park.</p>
+      </div>
+    );
+  }
+
+  return (
+    <div className="parks-wrapper">
+      <h1>{park.fullName}</h1>
+      <p>{park.description}</p>
+      <h2>Activities</h2>
+      <ul>
+        {park.activities.map((activity) => (
+          <li key={activity.id}>{activity.name}</li>
+        ))}
+      </ul>
+    </div>
+  );
```

Comment:
I wonder if you could DRY this up a little instead of having the double returns?

Maybe having a variable outside of the return that either holds "No activites..." or the results of the mapping.  Then you render that variable in the return?

Reply:
Hello Heather,

Thank you for your feedback! I wanted to let you know that the code you mentioned in your previous comment has been refactored and moved to the `ParkDetails` component as part of the ongoing development process. 

In `ParkDetails`, we now handle the rendering of park activities based on whether they are available or not. 
```
if (park === null) {
    return <LoadingComponent />;
  }
  if (!park) {
    return <ErrorComponent error={{ message: "Park information is not available." }} />;
  }
  
  const randomImage = park && park.images[Math.floor(Math.random() * park.images.length)];
  return (
    <div className='park-details'>
      <div className='button-container'>
        <Link to='/' className='back-button'>
          Back to Home
        </Link>
      </div>
      <div className='image-section'>
        <RandomImage randomImage={randomImage} />
        <h1 className='park-description'>{park.fullName}</h1>
        <div className='top-section-container'>
          <p>
            <strong>Location:</strong> {park.states}
          </p>
          <p>{park.description}</p>
          <div className='lower-section-container'>
            <div className='left-side'>
              <Activities activities={park.activities} />
              <AdditionalInfo park={park} />
            </div>
            <div className='right-side'>
              <ContactUs park={park} />
              <EntranceFees
                fees={park.entranceFees}
                toggleVisibility={toggleFeeVisibility}
                visibleFees={visibleFees}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
```

As always Thank You for your guidance and support and patience!

Best regards,
Parvin